### PR TITLE
Tidy up and make config page easier to read

### PR DESCRIPTION
**Pull request by rmoff**
* Code formatting 
* Split config into sections
* Add H2/H3 hierarchy to examples
* Remove duplicated env variables info

### CODE REVIEW

File: docs/reference/configuration.md [hunk 1]
```diff
@@ -1,75 +1,99 @@
...
-Configuration items can each be controlled by an environment variable. The variable name will have a prefix of *LAKEFS_*, followed by the name of the configuration, replacing every '.' with a '_'.
-Example: `LAKEFS_LOGGING_LEVEL` controls `logging.level`.
+The configuration file's location can be set with the `--config` flag. If not specified, the first file found in the following order will be used:
+1. `./config.yaml`
+1. `$HOME/lakefs/config.yaml`
+1. `/etc/lakefs/config.yaml`
+1. `$HOME/.lakefs.yaml`
 
-This reference uses `.` to denote the nesting of values.
+All the configuration variables can be set or overridden using environment variables. The variable name will have a prefix of *`LAKEFS_`*, followed by the name of the configuration, replacing every '`.`' with a '`_`'.
```
Comment (johnnyaug) [inline docs/reference/configuration.md:23]:
```suggestion
All the configuration properties can be set or overridden using environment variables. The variable name should have a prefix of *`LAKEFS_`*, followed by the name of the configuration, replacing every '`.`' with a '`_`'.
```

File: docs/reference/configuration.md [hunk 2]
```diff
@@ -178,28 +211,19 @@ This reference uses `.` to denote the nesting of values.
...
-## Using Environment Variables
-
-All the configuration variables can be set or overridden using environment variables.
-To set an environment variable, prepend `LAKEFS_` to its name, convert it to upper case, and replace `.` with `_`:
-
-For example, `logging.format` becomes `LAKEFS_LOGGING_FORMAT`, `blockstore.s3.region` becomes `LAKEFS_BLOCKSTORE_S3_REGION`, etc.
 
 
-## Example: Local Development with PostgreSQL database
+## Example lakeFS Server Configuration
```
Comment (johnnyaug) [inline docs/reference/configuration.md:225]:
```suggestion
## Examples
```

File: docs/reference/configuration.md [hunk 3]
```diff
@@ -1,75 +1,99 @@
 ---
 layout: default
-title: Configuration
-description: Configuring lakeFS is done using a YAML configuration file. This reference uses `.` to denote the nesting of values.
+title: lakeFS Server Configuration
```
Comment (johnnyaug) [inline docs/reference/configuration.md:3]: "lakeFS" here is redundant. We can go with "Server Configuration", "Configuring the Server" or leave the previous title.

Comment (rmoff) [reply docs/reference/configuration.md:3]: ack, thanks.

**Review by adipolak — APPROVED** [review]
LGTM.
those plugins could benefit from a sample app/lakeFS-sample repository addition.
Please open a ticket on devex and we will address it in the backlog.

**Pull request merged**